File: backend/src/Financas.Api/Domain/Models/NaturezaDeLancamento.cs
using System.ComponentModel.DataAnnotations;

namespace Financas.Api.Domain.Models
{
    public class NaturezaDeLancamento
    {
        [Key]
        public long Id { get; set; }

        [Required]
        [Range(1, long.MaxValue, ErrorMessage = "O ID do usuário deve ser maior que zero.")]
        public long IdUsuario { get; set; }
        public Usuario Usuario { get; set; }

        [Required(ErrorMessage = "O campo de descrição é obrigatório.")]
        public string Descricao { get; set; } = string.Empty;

        public string Observacao { get; set; } = string.Empty;

        [Required]
        public DateTime DataCadastro { get; set; }
        public DateTime? DataInativacao { get; set; }
    }
}

File: frontend/financas/src/models/NaturezaDeLancamento.js

export default class NaturezaDeLancamento {
    constructor(obj = {}) {
      this.id = obj.id;
      this.idUsuario = obj.idUsuario;
      this.descricao = obj.descricao;
      this.observacao = obj.observacao;
      this.dataCadastro = obj.dataCadastro;
      this.dataInativacao = obj.dataInativacao;
    }


    modeloValidoParaCadastro() {
      return !!(this.descricao);
    }
  
    modeloValidoParaAtualizar() {
      return !!(this.descricao);
    }
}
  